src/libraries/Fuzzing/DotnetFuzzing/MemoryBackedStream.cs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DotnetFuzzing;
+
+public class MemoryBackedStream : Stream
```

Comment:
Why not just use `MemoryStream`?

Reply:
Added a comment for usage. It's useful with the bounded memory test utility that gives a slice of virtual memory from the OS which is surrounded (either before or after) by a protected page so out-of-bounds accesses that aren't detected by CLR will be caught by the OS.

Reply:
I would assume this could only happen if stream implementation uses unsafe code. But even if that was the case, it would indicate a bug in the stream implementation and not the system under test.